mitmproxy/proxy/mode_specs.py

```diff
@@ -209,6 +209,52 @@ def __post_init__(self) -> None:
         _check_empty(self.data)
 
 
+class WireGuardMode(ProxyMode):
+    default_port = 51820
+    transport_protocol: ClassVar[Literal["tcp", "udp"]] = "udp"
+
+    wireguard_conf: str | None = None
+    wireguard_peers: int | None = None
+
+    # noinspection PyDataclass
+    def __post_init__(self) -> None:
+        # settings are empty or a literal "default":
+        if self.data in ["", "default"]:
+            return
+
+        # settings are comma-separated key-value pairs:
+        settings = self.data.split(",")
+
+        # override defaults with specified settings:
+        for kv in settings:
+            if not kv:
+                # empty string after a trailing comma
+                break
+
+            try:
+                key, value = kv.split("=", 1)
+            except ValueError:
+                raise ValueError(f"invalid wireguard mode (not a key=value pair: {kv!r})")
+
+            if key not in ["name", "listen_port", "peers"]:
+                raise ValueError(f"invalid wireguard mode (unrecognized option: {key!r}")
+
+            if key == "name":
+                self.wireguard_conf = value
+
+            if key == "listen_port":
```

Comment:
Ah, I see now that generic parsing of the mode spec happens in `ProxyMode.parse` ... This might conflict with the custom format that's used by WireGuardMode right now - might explain why apparently mitmproxy gets fed an invalid IP address when you try to run it :)

Looks like I shouldn't use `:` as separator character in `wireguard:custom=foo,settings=bar,` after all, because this breaks things elsewhere - any suggestions for what I could use instead?

Reply:
`:` should be fine, but if `@` appears anywhere in your settings, the string after `@` will be interpreted as `[host:]port`, which will fail.

@mhils: Maybe we should use regex to ensure `@` only matches if it's followed by an IP and port? (Of course for something like `wireguard:conf_prefix=@7` this would still be a problem, but only people named [**Robert'); DROP TABLE Student;--**](https://xkcd.com/327/) pick a prefix like that ;)

Reply:
> Looks like I shouldn't use `:` as separator character

I think you need to, that's what ProxyMode.parse expects.

> @mhils: Maybe we should use regex to ensure @ only matches if it's followed by an IP and port?

Let's just not support `@` in specs for now, I don't think it's much of a restriction. :)

Reply:
Yeah, `@` characters aren't expected in any of the settings that the wireguard mode will accept.

Still doesn't help that I have no idea where the `invalid IP address syntax` error comes from then :/

Reply:
I simplified the mode spec a bit based on feedback here, and by adding better test coverage, I found the error (it's not in my code, but in ProxyMode.parse. For a mode like `wireguard@51821:generate`, it will attempt to parse `51821` as a host address and `generate` as a port number.

Reply:
The spec is invalid, it should be `wireguard:generate@51821` in your example. The `@listen_port` part always comes last.

Reply:
Oh, that finally makes sense. I had misinterpreted the format.